src/screens/Search/SearchScreen/index.js

```diff
@@ -30,55 +30,43 @@ class SearchScreen extends Component {
       manga: [],
       users: [],
     },
-    index: 0,
-    routes: [
-      {
-        key: 'anime',
-        title: 'Anime',
+    scenes: {
+      anime: {
         apiKey: this.props.algoliaKeys.media && this.props.algoliaKeys.media.key,
         indexName: this.props.algoliaKeys.media && this.props.algoliaKeys.media.index,
         kind: 'anime',
       },
-      {
-        key: 'manga',
-        title: 'Manga',
+      manga: {
         apiKey: this.props.algoliaKeys.media && this.props.algoliaKeys.media.key,
         indexName: this.props.algoliaKeys.media && this.props.algoliaKeys.media.index,
         kind: 'manga',
       },
-      {
-        key: 'users',
-        title: 'Users',
+      users: {
         apiKey: this.props.algoliaKeys.users && this.props.algoliaKeys.users.key,
         indexName: this.props.algoliaKeys.users && this.props.algoliaKeys.users.index,
       },
-    ],
+    }
   };
 
-  doSearch = (query, route) => {
-    if (!route || !route.apiKey) return;
-
-    const algoliaClient = algolia(kitsuConfig.algoliaAppId, route.apiKey);
-    const algoliaIndex = algoliaClient.initIndex(route.indexName);
-
-    const filters = route.kind ? `kind:${route.kind}` : '';
+  handleQuery = (scene, query) => {
+   const nextState = { ...this.state.query, [scene]: isEmpty(query) ? undefined : query };
+   this.setState({ query: nextState }, () => {
+     this.debouncedSearch(query, scene);
+   });
+  };
 
-    algoliaIndex.search({ query, filters }, (err, content) => {
+  executeSearch = (query, scene) => {
+    const currentScene = this.state.scenes[scene];
+    const client = algolia(kitsuConfig.algoliaAppId, currentScene.apiKey);
+    const index = client.initIndex(currentScene.indexName);
+    const filters = currentScene.kind ? `kind:${currentScene.kind}` : '';
+    index.search({ query, filters}, (err, content) => {
       if (!err) {
-        this.setState({ searchResults: { [route.key]: content.hits } });
+        this.setState({ searchResults: { [scene]: content.hits } });
       }
     });
   };
-
-  handleSearchStateChange = (route, query) => {
-    // const { query } = searchState;
-    const nextQueryState = { ...this.state.query, [route.key]: query !== '' ? query : undefined };
-    this.setState({ query: nextQueryState }, () => {
-      this.doSearch(query, route);
-    });
-  };
-
-  handleIndexChange = index => this.setState({ index });
+  debouncedSearch = debounce(this.executeSearch, 250);
```

Comment:
I'd lower this to 100-200ms, honestly. Algolia is quick enough that we don't need as much debouncing and we make the search feel quicker by reducing this